From 5c76751daebcb45d2d9a5d96ad7a69f2eae802b1 Mon Sep 17 00:00:00 2001
From: robinz1x <robinx.zhang@intel.com>
Date: Thu, 21 Jun 2018 14:50:00 +0800
Subject: [PATCH] Fix java crash in CarLatinIME.updateCapitalization

[Port from Gordon_Peak]
The inputConnection object maybe invalid if the previous
bound client is no longer associated with the input method,
so add this object reference non-null check in updateCaptialization
when get and use inputConnection object.

Change-Id: I7d58d1e586209c495c1d28c6504fdc37127e9f2d
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-65692
Signed-off-by: robinz1x <robinx.zhang@intel.com>
---
 src/com/android/inputmethod/latin/CarLatinIME.java | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/com/android/inputmethod/latin/CarLatinIME.java b/src/com/android/inputmethod/latin/CarLatinIME.java
index 861fd8d..8024a76 100644
--- a/src/com/android/inputmethod/latin/CarLatinIME.java
+++ b/src/com/android/inputmethod/latin/CarLatinIME.java
@@ -242,9 +242,12 @@ public class CarLatinIME extends InputMethodService {
     }
 
     private void updateCapitalization() {
-        boolean shouldCapitalize =
-                getCurrentInputConnection().getCursorCapsMode(mEditorInfo.inputType) != 0;
-        mKeyboardView.setShifted(shouldCapitalize);
+        InputConnection inputConnection = getCurrentInputConnection();
+        if (inputConnection != null) {
+            boolean shouldCapitalize =
+                inputConnection.getCursorCapsMode(mEditorInfo.inputType) != 0;
+            mKeyboardView.setShifted(shouldCapitalize);
+        }
     }
 
     private final KeyboardView.OnKeyboardActionListener mKeyboardActionListener =
-- 
1.9.1

